### PR TITLE
spice-gtk: 0.40 -> 0.41, add darwin support

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

https://gitlab.freedesktop.org/spice/spice-gtk/-/blob/v0.41/CHANGELOG.md

Supersedes #183689 and #183674.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/spice-gtk/default.nix [hunk 1]
```diff
@@ -16,7 +17,7 @@
 , libdrm
 , libjpeg_turbo
 , libopus
-, libsoup
+, libsoup_3
```
Comment (jtojnar) [inline pkgs/development/libraries/spice-gtk/default.nix:20]: The libsoup 3 migration unfortunately is not that simple. It needs to be switched in all the reverse dependencies at the same time, see https://discourse.gnome.org/t/phodav-transition-notes/10483 and #182618

Comment (wegank) [reply pkgs/development/libraries/spice-gtk/default.nix:20]: I see. I'll push another commit that reverts to libsoup.

Comment (wegank) [reply pkgs/development/libraries/spice-gtk/default.nix]: Reverted to libsoup now.

**Comment by wegank** [comment]
@ofborg eval

File: pkgs/tools/networking/phodav/default.nix [hunk 2]
```diff
@@ -12,13 +12,26 @@ in stdenv.mkDerivation rec {
     sha256 = "045rdzf8isqmzix12lkz6z073b5qvcqq6ad028advm5gf36skw3i";
   };
 
+  patches = lib.optionals stdenv.isDarwin [
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/phodav/default.nix:15]:
```suggestion
  patches = [
```

File: pkgs/tools/networking/phodav/default.nix [hunk 3]
```diff
@@ -28,6 +41,6 @@ in stdenv.mkDerivation rec {
     homepage = "https://wiki.gnome.org/phodav";
     license = licenses.lgpl21;
     maintainers = with maintainers; [ ];
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/phodav/default.nix:43]: Maybe consider adding yourself as maintainer here or the other packages

File: pkgs/tools/networking/phodav/default.nix (hunk 2 above)
Comment (wegank) [reply pkgs/tools/networking/phodav/default.nix]: Done.

File: pkgs/tools/networking/phodav/default.nix (hunk 3 above)
Comment (wegank) [reply pkgs/tools/networking/phodav/default.nix]: Done.

**Pull request merged**